custom/icds_reports/reports/governance_apis.py

```diff
@@ -38,3 +39,56 @@ def base_data(row_data):
     for row in paginated_data:
         data_rows.append(base_data(row))
     return data_rows, data.count()
+
+
+@icds_quickcache(['start', 'length', 'year', 'month', 'order', 'query_filters'], timeout=30 * 60)
+def get_beneficiary_data(start, length, year, month, order, query_filters):
+    data = AggAwcMonthly.objects.filter(
```

Comment:
I would like to see a comparison between this and using a JOIN or even two queries with AggChildHealth and AggCcsRecord. This table is getting wildly too large, and we should do our best to only add columns if absolutely necessary.

Reply:
Hey, I think that would be a bad idea. because we would just want to want launched columns. We have tried joining these three tables in past in THR report which gave a worst performance with all-day running queries. That's why I thought of adding the columns here.

Reply:
I could also do joining in Python after fetching data separately from three tables. But not sure how good that would be,

Reply:
This takes our widest table by a large margin and makes it 10% wider. At the current rate, eventually that table will just include all columns from each of the other agg tables, which is not a feasible state. We need to find a way to add columns to these APIs without just putting them in AggAwc. That might mean a new table for this API, if necessary, or since we are limiting the size of the queries so much the join may not be as unperformant (the bad THR ones were both worse queries and entire states, which are prevented in this API).

Reply:
Alternatively, as you mentioned, you can do it in python. It might be worth benchmarking both approaches.

Reply:
Hey, I think at this point I think it to create a new table, not the joins. I tried, the cost hardly reduces with the query limits. Also querying them separately then joining the result in python would just take longer because anyway we need to query agg_Awc table and along with that we need to query the agg_Child whose query time is even higher than agg_awc. and we can not query the child health table because it gives 10-15 minutes query completion time historically of similar queries.

YOur thoughts, comments, suggestions?

Reply:
Also, I am curious about your concern to be clear about adding the columns to agg_awc. Are you concern in terms of the future state of a massive table and we will run out of limit. or you are saying adding more and more columns to table slows it down?

@calellowitz